pandas/core/indexes/base.py

```diff
@@ -2922,6 +2922,9 @@ def is_int(v):
                     self._validate_indexer("slice", key.step, kind),
                 )
 
+        if kind == "loc" and not is_null_slicer:
+            return self.slice_indexer(key.start, key.stop, key.step, kind=kind)
+
```

Comment:
is any of the validation on L2931-2948 relevant?  can some or all of it be removed?

This likely affects some tests, see how this is handled in #31840.

Reply:
hey sorry I haven't had time to work on this.
will close this PR in favor of #31840.